crates/chia-puzzles/src/derive_synthetic.rs

```diff
@@ -118,7 +118,7 @@ mod tests {
         for (index, hex) in hex_keys.iter().enumerate() {
             let key = intermediate
                 .derive_unhardened(index as u32)
-                .derive_synthetic(&DEFAULT_HIDDEN_PUZZLE_HASH);
+                .derive_synthetic(&DEFAULT_HIDDEN_PUZZLE_HASH.to_bytes());
```

Comment:
are there any circumstances where having this be a `TreeHash` improves something? e.g. ergonomics or type-safety

Reply:
Yeah, the idea is you can pass a puzzle hash directly into a generic inner puzzle field, and it will act as pre-hashed (ie as if you put the preimage there instead). If it were `[u8; 32]` you could easily mistakenly curry in a list of 32 bytes instead of the hash as-is.